kernel/names.ml

```diff
@@ -368,14 +368,19 @@ module KerName = struct
   type t = {
     modpath : ModPath.t;
     knlabel : Label.t;
-    mutable refhash : int;
+    refhash : int;
     (** Lazily computed hash. If unset, it is set to negative values. *)
   }
 
   type kernel_name = t
 
   let make modpath knlabel =
-    { modpath; knlabel; refhash = -1; }
+    let open Hashset.Combine in
+    let refhash = combine (ModPath.hash modpath) (Label.hash knlabel) in
+    (* Ensure positivity on all platforms. *)
+    let refhash = refhash land 0x3FFFFFFF in
```

Comment:
We don't need to ensure positivity if we're not using `-1` as a sentinel do we?

Reply:
I think the comment is misleading, it's there not so much for positivity but for portability between platforms. At some point we tried to remove the sources of divergence for vo files between 32 and 64-bit architectures, but AFAIK we've given up on that.